### ISSUE TITLE
Modify the way Error is displayed under FAILED tab

### ISSUE THREAD

**Issue by mannharleen** [felixmosh/bull-board#230]
## What I noticed:
When `error.message` (nodejs error object) is an object rather than a string, the UI doesn't handle the display properly. (see image below): 
![image](https://user-images.githubusercontent.com/22973248/108064923-6a4ebf00-70b1-11eb-9a57-d89c58446332.png)

## Possible solution options
We could fix this on the UI, however, I also feel that showing stack traces may not be acceptable at certain times. From the security point of view, I dont want my stack traces to be shown by default unless asked for. This could be handled as a separately feature (a button maybe!?).

## My suggested solution - 
I believe, instead of showing `stacktrace` from the redis job, we should instead show the `failedReason` field.

### Looking forward to your comments...

> If my suggestion is acceptable, I am happy to submit a PR.

**Comment by felixmosh** [comment]
You can catch the failure of the job & re throw a custom object.
This checks that if there is no `stacktrace` && there is `failedReason`, it will show `failedReason`.
https://github.com/vcapretz/bull-board/blob/master/src/ui/hooks/useDetailsTabs.tsx#L49

Let me know if this solves the issue

**Comment by mannharleen** [comment]
> You can catch the failure of the job & re throw a custom object.
> This checks that if there is no `stacktrace` && there is `failedReason`, it will show `failedReason`.
> https://github.com/vcapretz/bull-board/blob/master/src/ui/hooks/useDetailsTabs.tsx#L49
> 
> Let me know if this solves the issue

Thanks for the reference. So It turns out that `stackTrace` item that bullmq stored on redis, is actually always an array (it stored more information than just the actual stacktrace). `job.moveToFailed` expects an error as the first argument. So I tried the following combinations:
```
1. e = New Error('xxx'); e.stack= null
2. e = New Error('xxx'); e.stack= ''
3. 2. e = New Error('xxx'); delete e.stack
Each one of the above resulted in an array in `job.stackTrace` in redis:
1 resulted in => [null]
2 resulted in => [""]
3 resulted in => [null]
```
So, `stacktrace?.length === 0` is never true ! 

> Do we have test cases for stacktrace?.length === 0 ?

My workaround for now is to create a new error object and mark stack = message
`eNew.stack = e.message`

**Comment by felixmosh** [comment]
> > You can catch the failure of the job & re throw a custom object.
> > This checks that if there is no `stacktrace` && there is `failedReason`, it will show `failedReason`.
> > https://github.com/vcapretz/bull-board/blob/master/src/ui/hooks/useDetailsTabs.tsx#L49
> > Let me know if this solves the issue
> 
> Thanks for the reference. So It turns out that `stackTrace` item that bullmq stored on redis, is actually always an array (it stored more information than just the actual stacktrace). `job.moveToFailed` expects an error as the first argument. So I tried the following combinations:
> 
> ```
> 1. e = New Error('xxx'); e.stack= null
> 2. e = New Error('xxx'); e.stack= ''
> 3. 2. e = New Error('xxx'); delete e.stack
> Each one of the above resulted in an array in `job.stackTrace` in redis:
> 1 resulted in => [null]
> 2 resulted in => [""]
> 3 resulted in => [null]
> ```
> 
> So, `stacktrace?.length === 0` is never true !
> 
> > Do we have test cases for stacktrace?.length === 0 ?
> 
> My workaround for now is to create a new error object and mark stack = message
> `eNew.stack = e.message`

I'll create a PR that resolves this issue of array with null or ''

**Comment by mannharleen** [comment]
I am happy to create the PR, if you d like me to instead.

**Comment by felixmosh** [comment]
I've released the #235 fix in v1.3.1 can you check if this suits your need?

**Comment by mannharleen** [comment]
all good.
